interfaces/repo.go

```diff
@@ -283,19 +283,24 @@ func (r *Repository) Connection(connRef *ConnRef) (*Connection, error) {
 	// Ensure that such plug exists
 	plug := r.plugs[connRef.PlugRef.Snap][connRef.PlugRef.Name]
 	if plug == nil {
-		return nil, fmt.Errorf("snap %q has no plug named %q", connRef.PlugRef.Snap, connRef.PlugRef.Name)
+		return nil, &NoPlugOrSlotError{
+			message: fmt.Sprintf("snap %q has no plug named %q",
+				connRef.PlugRef.Snap, connRef.PlugRef.Name)}
 	}
 	// Ensure that such slot exists
 	slot := r.slots[connRef.SlotRef.Snap][connRef.SlotRef.Name]
 	if slot == nil {
-		return nil, fmt.Errorf("snap %q has no slot named %q", connRef.SlotRef.Snap, connRef.SlotRef.Name)
+		return nil, &NoPlugOrSlotError{
```

Comment:
same,  also let's try to be consistent and adopt these errors in - afaict:

* connected
* ResolveConnect
* Disconnect

Reply:
Done